### PR TITLE
my Hello World submission

### PR DESCRIPTION
**Pull request by mwaldl**

### CODE REVIEW

**Comment by s-will** [comment]
Der Programmname muss genau umgekehrt zusammengesetzt sein: mwaldl-HelloWorld.cc

**Comment by mwaldl** [comment]
any suggestions on c++ style guides are welcome :)

File: A0/mwaldl-HelloWorld.cc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+main( int argc, char *argv[] ) {
+    
+  // print "Hello World!" + new line
+  cout << "Hello World!\n";
+
+  // print content of a file to standard out
+  ifstream file ( argv[1] ); // assuming the second command line argument is a file path
+  if ( file.is_open() ) {    
+    cout << file.rdbuf();
```
Comment (s-will) [inline A0/mwaldl-HelloWorld.cc]: a concise way to do this

File: submissions.yml [hunk 2]
```diff
@@ -23,6 +23,14 @@ submissions:
 
             # set 'checked' to true to turn off further ci tests
             checked:  false
+        mwaldl:
+            # define the language of the submission (see
+            # languages.yml); this controls whether and how the
+            # program is compiled and how it is invoked
```
Comment (s-will) [inline submissions.yml]: actually no need to duplicate my comment; probably better to clean this

**Comment by s-will** [comment]
Hi @mwaldl, congrats you were the first with a working solution.

File: A0/mwaldl-HelloWorld.cc [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  } else {
+    cout << "Could not open file\n";
+  }
+  return 0;
+}
```
Comment (mwaldl) [inline A0/mwaldl-HelloWorld.cc]: not sure if I understand this: "file" is part of a string, not a variable name. Of course I could use  '<<endl' in stead of '\n' to flush the buffer.

Comment (s-will) [inline A0/mwaldl-HelloWorld.cc]: (as also github indicated)  there is no newline at the end of your program text, i.e. after the last '}' in line 22.

File: A0/mwaldl-HelloWorld.cc (hunk 1 above)
Comment (mwaldl) [inline A0/mwaldl-HelloWorld.cc]: should I use something like  os.flush() here too?

File: A0/mwaldl-HelloWorld.cc (hunk 3 above)
Comment (mwaldl) [inline A0/mwaldl-HelloWorld.cc]: ah - I was looking at the wrong line
thanks

**Comment by s-will** [comment]
enough is enough :)

**Pull request merged**